src/plugins/profiling/server/routes/load_topn.ts

```diff
@@ -0,0 +1,145 @@
+          index: schema.maybe(schema.string()),
+          projectID: schema.maybe(schema.string()),
+          timeFrom: schema.maybe(schema.string()),
+          timeTo: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    async (ctx, request, response) => {
+      const [timeFrom, timeTo] = timeRangeFromRequest(request);
+      const src = await import(`../fixtures/containers_${timeTo - timeFrom}`);
+      delete src.default;
+      return response.ok({ body: src });
+    }
+  );
+}
+
+export function registerTraceEventsTopNDeploymentsRoute(
+  router: IRouter<DataRequestHandlerContext>
+) {
+  const paths = getLocalRoutePaths();
+  router.get(
+    {
+      path: paths.TopNDeployments,
+      validate: {
+        query: schema.object({
+          index: schema.maybe(schema.string()),
+          projectID: schema.maybe(schema.string()),
+          timeFrom: schema.maybe(schema.string()),
+          timeTo: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    async (ctx, request, response) => {
+      const [timeFrom, timeTo] = timeRangeFromRequest(request);
+      const src = await import(`../fixtures/pods_${timeTo - timeFrom}`);
+      delete src.default;
+      return response.ok({ body: src });
+    }
+  );
+}
+
+export function registerTraceEventsTopNHostsRoute(router: IRouter<DataRequestHandlerContext>) {
+  const paths = getLocalRoutePaths();
+  router.get(
+    {
+      path: paths.TopNHosts,
+      validate: {
+        query: schema.object({
+          index: schema.maybe(schema.string()),
+          projectID: schema.maybe(schema.string()),
+          timeFrom: schema.maybe(schema.string()),
+          timeTo: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    async (ctx, request, response) => {
+      const [timeFrom, timeTo] = timeRangeFromRequest(request);
+      const src = await import(`../fixtures/hosts_${timeTo - timeFrom}`);
+      delete src.default;
+      return response.ok({ body: src });
+    }
+  );
+}
+
+export function registerTraceEventsTopNStackTracesRoute(
+  router: IRouter<DataRequestHandlerContext>
+) {
+  const paths = getLocalRoutePaths();
+  router.get(
+    {
+      path: paths.TopNTraces,
+      validate: {
+        query: schema.object({
+          index: schema.maybe(schema.string()),
+          projectID: schema.maybe(schema.string()),
+          timeFrom: schema.maybe(schema.string()),
+          timeTo: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    async (ctx, request, response) => {
+      const [timeFrom, timeTo] = timeRangeFromRequest(request);
+      const src = await import(`../fixtures/traces_${timeTo - timeFrom}`);
+      delete src.default;
+      return response.ok({ body: transformFlamechart(src) });
+    }
+  );
+}
+
+export function registerTraceEventsTopNThreadsRoute(router: IRouter<DataRequestHandlerContext>) {
+  const paths = getLocalRoutePaths();
+  router.get(
+    {
+      path: paths.TopNThreads,
```

Comment:
Does it make sense to reduce code duplication ? E.g. with a function that takes 'router', 'path' and 'fixture_prefix` ?

Reply:
yeah I had thought about it 🤔